pkg/cmd/release/update/update.go

```diff
@@ -0,0 +1,186 @@
+package update
+
+import (
+	"errors"
+	"fmt"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+	"net/http"
+)
+
+type UpdateOptions struct {
+	IO         *iostreams.IOStreams
+	HttpClient func() (*http.Client, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	ReleaseId    string
+	TagName      string
+	Target       string
+	Name         string
+	Body         string
+	BodyProvided bool
+	Draft        *bool
+	Prerelease   *bool
+
+	// the value from the --repo flag
+	RepoOverride string
+}
+
+func NewCmdUpdate(f *cmdutil.Factory, runF func(*UpdateOptions) error) *cobra.Command {
+	opts := &UpdateOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	var notesFile string
+
+	cmd := &cobra.Command{
+		DisableFlagsInUseLine: true,
+
+		Use:   "update <release_id>",
+		Short: "Edit a release",
+		Example: heredoc.Doc(`
+			Publish a release that was previously a draft
+			$ gh release update 63899317 --draft=false
+
+			Mark a release as a prerelease
+			$ gh release update 63899317 --prerelease
+
+			Mark a release as a full release
+			$ gh release update 63899317 --prerelease=false
+
+			Update the target commitish of a release
+			$ gh release update 63899317 --target 97ea5e77b4d61d5d80ed08f7512847dee3ec9af5
+
+			Update the name of the tag
+			$ gh release update 63899317 --tag v9.8.7
+
+			Update the name/title of a release
+			$ gh release update 63899317 --title 'Some new title'
+
+			Update the release notes
+			$ gh release update 63899317 --notes 'Some short release notes'
+
+			Update the release notes from the content of a file
+			$ gh release update 63899317 --notes-file /path/to/release_notes.md
```

Comment:
These docs are great, but we don't necessarily have to demonstrate the use of every single supported flag. Passing string values should be self-evident and most of them we do not have to document. However, we could document passing notes from file, like you did here, and demonstrate the `--draft=false` syntax for booleans.